src/App.tsx

```diff
@@ -1,56 +1,78 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import { debounce, getFilteredPeople } from './helpers/Functions';
+import { DropMenu } from './components/DropMenu';
+import { Input } from './components/Input';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [personName, setPersonName] = useState('');
+  const [isFocused, setIsFocused] = useState(false);
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [people, setPeople] = useState<Person[]>([...peopleFromServer]);
+  const [isDropdownVisible, setDropdownVisible] = useState(true);
+  const [selectedPersonSlug, setSelectedPersonSlug] = useState('');
+
+  const applyQuery = debounce(setAppliedQuery, 1000);
```

Comment:
As you can see the filter function `getFilteredPeople` filter array of people several times depending on amount of symbols in your input. You need to prevent it and filter array of people after 1 second. You should wrap your debounce function in `useCallback` to prevent prevent re-rendering of the function
```suggestion
  const applyQuery = useCallback(debounce(setAppliedQuery, 1000), []);
```
Don't forget to import it from react